packages/lab/src/carousel/CarouselSlider.tsx

```diff
@@ -0,0 +1,99 @@
+import { makePrefixer, useForkRef } from "@salt-ds/core";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import {
+  type HTMLAttributes,
+  type KeyboardEvent,
+  type ReactElement,
+  forwardRef,
+  useCallback,
+} from "react";
+import { useIntersectionObserver } from "../utils";
+import { type CarouselContextValue, useCarousel } from "./CarouselContext";
+import type { CarouselSlideProps } from "./CarouselSlide";
+import carouselSliderCss from "./CarouselSlider.css";
+
+export interface CarouselSliderProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * Collection of slides to render
+   */
+  children: Array<ReactElement<CarouselSlideProps>>;
+}
+
+const withBaseName = makePrefixer("saltCarouselSlider");
+
+const useKeyNavigation = ({
+  nextSlide,
+  prevSlide,
+}: Pick<CarouselContextValue, "nextSlide" | "prevSlide">) => {
+  return useCallback(
+    (event: KeyboardEvent<HTMLDivElement>) => {
+      if (event.key === "ArrowRight" || event.key === "ArrowLeft") {
+        if (event.repeat) return;
+        event.stopPropagation();
+        if (event.key === "ArrowRight") {
+          nextSlide?.(event);
+        } else if (event.key === "ArrowLeft") {
+          prevSlide?.(event);
+        }
+      }
+    },
+    [nextSlide, prevSlide],
+  );
+};
+
+export const CarouselSlider = forwardRef<HTMLDivElement, CarouselSliderProps>(
+  function CarouselSlider(
+    { children, onKeyDown: onKeyDownProp, ...rest },
+    propRef,
+  ) {
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-carousel-slide",
+      css: carouselSliderCss,
+      window: targetWindow,
+    });
+
+    const {
+      updateActiveFromScroll,
+      containerRef,
+      prevSlide,
+      nextSlide,
+      visibleSlides,
+    } = useCarousel();
+    const handleKeyDown = useKeyNavigation({
+      nextSlide,
+      prevSlide,
+    });
+
+    // Handlers
+    const handleScroll = useCallback(() => {
+      if (containerRef.current) {
+        const scrollLeft = containerRef.current.scrollLeft;
+        updateActiveFromScroll(scrollLeft);
+      }
+    }, [containerRef, updateActiveFromScroll]);
```

Comment:
We offer an uncontrolled component but what if we offered a controlled component then folks could do alot of things themselves.

For instance, they could create an autoPlay  and we could document how to create it ?

Most other Carousels provide an API that provide events for slides becoming visible ?